apps/examples/smartfs/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -175,18 +179,41 @@ static int write_text_file(int fd, size_t bufsize)
 	return ret;
 }
 
-static int create_file(char *path, ssize_t bufsize, file_type_t type)
+static int validate_file(char *path, ssize_t bufsize)
 {
-	int fd;
 	int ret;
 	struct stat st;
 
-	/* If file is already exist and written fully, just return OK */
+	/* Check file existence and size and return accordingly */
 	if (stat(path, &st) == OK) {
 		printf("File already exist path : %s size : %d\n", path, st.st_size);
-		return OK;
+
+		/* If the complete file exists, return accordingly
+		 * Do not unlink if it is a main file, as it will be recovered later
+		 * Don't need to recreate a backup file that is complete either.
+		 */
+		if (IS_MAIN_FILE(path) || st.st_size == (bufsize * TEST_WRITECOUNT)) {
+			return -EEXIST;
+		}
+
+		/* The backup file does not exist completely, unlink and recreate */
+		ret = unlink(path);
+		if (ret != OK) {
+			printf("Unable to unlink incomplete backup file: %s, ret: %d, errno: %d\n", path, ret, errno);
+			return ERROR;
+		}
+		printf("Incomplete backup file: %s unlinked\n", path);
 	}
-	
+
+	/* Return value of OK will indicate that file creation should be done after this function returns */
+	return OK;
+}
+
+static int create_file(char *path, ssize_t bufsize, file_type_t type)
```

Comment:
1. only backup file need to be checked.
2. We should check crc of each of data chunk in backup file too, so we should use verify_file.
```
       int index = TEST_WRITECOUNT;
       int ret = verify_file(TEST_SMALL_R_BACKUP, TEST_SMALL_BUFSIZE, &index, FILE_TYPE_RANDOM)
       if (ret == OK && index != TEST_WRITE_COUNT) {
           unlink();
        } else {
      	   ret = create_file(TEST_SMALL_R_BACKUP, TEST_SMALL_BUFSIZE, FILE_TYPE_RANDOM, true);
           if (ret != OK && errno != EEXIST) {
		return ERROR;
           }
        }
```
3. but creat file already check file is exist or not through the stat, you should optimize it.

Reply:
1. Init is supposed to create all test files, so validate tells us if we need to create new file or it already exists.

2.  We dont need to check the contents of the backup file because it is either renamed from main file which was checked OR it was created during init. So the contents will be OK as journaling is enabled and recovery happens at bootup.
If the size is incomplete, we do not truncate and extend it, we create the whole file again.
Also later, the mail file will be renamed to backup and main file is verified properly.

3. Removed the checks from create_file and put them into validate_file so that create_file only does what the name suggests.

Reply:
> 1. Init is supposed to create all test files, so validate tells us if we need to create new file or it already exists.
=> I don't think so, init_file will be called for every time, in this case we should not check contents file to verify&recovery
> 2. We dont need to check the contents of the backup file because it is either renamed from main file which was checked OR it was created during init. So the contents will be OK as journaling is enabled and recovery happens at bootup.
=> ah yes, agree with your point.
>    If the size is incomplete, we do not truncate and extend it, we create the whole file again.
>    Also later, the mail file will be renamed to backup and main file is verified properly.
> 3. Removed the checks from create_file and put them into validate_file so that create_file only does what the name suggests.
 => I though I commented mine in 'init_test_file'

Reply:
so regarding point #1, you don't have to use MACRO 'IS_MAIN_FILE'

Reply:
If backup file is not exist or valid, then it should return OK. Otherwise, unlink backup file & return ERROR.
Then,
```
ret = is_valid_backup_file(); // if it is not valid, it will unlink file, so we don't have to check -EEXIST anymore.
if (ret == ERROR) {
     ret = create_file();
}
return ret; //so just return ret itself,